rss/lambda/lambda.py

```diff
@@ -0,0 +1,69 @@
+import os
+import sys
+import time
+import json
+import botocore, boto3
+
+# This is stupid, but you can't explicitly import module dependencies.
+sys.path.append('./libs')
+
+import feedparser
+import requests
+
+
+def handler(event, context):
+    db = boto3.resource(
+        "dynamodb",
+        region_name=os.environ["dynamo_region"],
+        config=botocore.client.Config(),
+    ).Table(os.environ["dynamo_table"])
+
+    response = db.query(
+        KeyConditionExpression="id = :id",
+        ExpressionAttributeValues={":id": "lambda.py"},
+    )
+    if len(response["Items"]) == 0:
+        return { "statusCode": 500, "body": json.dumps(f"database not configured"), }
+
+    firstResponse = response["Items"][0]
+    lastUpdateSeen = float(firstResponse["ts"])
+    sharedSecret = firstResponse["ss"]
+
+    d = feedparser.parse(
+        "https://htcondor-wiki.cs.wisc.edu/index.cgi/timeline.rss?d=90"
+    )
+    lastUpdate = time.mktime(d.feed.updated_parsed)
+
+    if lastUpdateSeen < lastUpdate:
+        db.update_item(
+            Key={"id": "lambda.py"},
+            UpdateExpression="set ts = :ts",
+            ExpressionAttributeValues={":ts": str(lastUpdate)},
+            ReturnValues="UPDATED_NEW",
+        )
+
+        newEntries = []
+        for entry in d.entries:
+            published = time.mktime(entry.published_parsed)
+            if lastUpdateSeen < published:
+                newEntries.append(entry)
+
+        requests.post(
+            "https://chtc-bot.herokuapp.com/rss",
```

Comment:
Can we pass the URL in through an environment variable? Thinking about future-proofing against changes in the bot hosting solution.

Reply:
Yes.